src/App.tsx

```diff
@@ -1,26 +1,210 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import * as todosFromServer from './api/todos';
+import { wait } from './utils/fetchClient';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { Status } from './types/Status';
+import { TodoItem } from './components/TodoItem/TodoItem';
 
-const USER_ID = 0;
+const getTodosByStatus = (status: string, todos: Todo[]) => {
+  const preperedTodos = [...todos];
+
+  if (status) {
+    switch (status) {
+      case Object.keys(Status)[Object.values(Status).indexOf(Status.active)]:
+        return preperedTodos.filter(todo => !todo.completed);
+      case Object.keys(Status)[Object.values(Status).indexOf(Status.completed)]:
+        return preperedTodos.filter(todo => todo.completed);
+      default:
+        return preperedTodos;
+    }
+  }
+
+  return preperedTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<Todo[]>([]);
+  const [edit, setEdit] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const temp = (currentTodo: Todo) => {
+    setLoadingTodos(prevArray => [...prevArray, currentTodo]);
+  };
+
+  const filteredTodos = getTodosByStatus(status, todos);
+
+  const addTodo = async (newTodoTitle: string) => {
+    const editedTitle = newTodoTitle.trim();
+
+    if (!editedTitle) {
+      setErrorMessage('Title should not be empty');
+      wait(3000).finally(() => setErrorMessage('Title should not be empty'));
```

Comment:
```suggestion
      setErrorMessage('Title should not be empty');
```
you already showed the message, no need to do it again after a delay